### ISSUE TITLE
Test mode misses some commands

### ISSUE THREAD

**Issue by FormerLurker** [FormerLurker/Octolapse#12]
Occationally I've noticed that my plugin might miss the first few GCodes sent to the printer.  This has only ever affected me while using the debugger, but I think it's possible that it could happen during a test print.  This can have the effect of letting commands that should be suppressed/rewritten slip past my routines and make it to the printer.  In the worst case these commands could heat up your bed or extruder.  When using test mode make sure you keep an eye on your printer's fans, temperature, and extruder motor and ALWAYS unload your filament to be safe.

There are probably some additional commands that should be suppressed/rewritten for test mode that are not included.  If you find any let me know.

**Comment by FormerLurker** [comment]
I have not had, nor heard of this problem in a while, so perhaps it is fixed.  I'll leave this up for a while.

**Comment by FormerLurker** [comment]
I've changed the event used to detect print start.  It seems as though PrinterStateChanged  fires before Print_Started.  This had been previously implemented, but some additions used to stop prints in the event of a non-local file (PrinterSD).  This has been moved now to Print_Started, and has been separated from the start up routine.

I tested pausing the debugger inside of the PrinterStateChanged event handler, and printing continues, so Octoprint is not waiting for my to return control.  To properly fix this I'll have to grab a lock immediately when the event is triggered.  To do this I need to update to the latest dev build of Octoprint.

I've tested my fix on gcode that had the home command on the first line, and it worked on my MK2S in test mode.
